frontend/components/modals/ShowQueryModal/ShowQueryModal.tsx

```diff
@@ -9,12 +11,31 @@ const baseClass = "show-query-modal";
 interface IShowQueryModalProps {
   onCancel: () => void;
   query?: string;
+  liveQuery?: boolean;
+  livePolicy?: boolean;
 }
 
 const ShowQueryModal = ({
   query,
+  liveQuery,
+  livePolicy,
   onCancel,
 }: IShowQueryModalProps): JSX.Element => {
+  const { lastEditedQueryBody } = useContext(QueryContext);
+  const { lastEditedQueryBody: lastEditedPolicyQueryBody } = useContext(
+    PolicyContext
+  );
+
+  const querySql = () => {
+    if (liveQuery) {
+      return lastEditedQueryBody;
+    }
+    if (livePolicy) {
+      return lastEditedPolicyQueryBody;
+    }
+    return query;
+  };
+
```

Comment:
Could you pass the query as a prop to this component? That would make it more reusable.

Reply:
Yes i would agree with this. Nice pulling this out into one component though!

Reply:
Yes! Thank you